src/segments/nixshell.go

```diff
@@ -0,0 +1,67 @@
+package segments
+
+import (
+	"path/filepath"
+	"strings"
+
+	"github.com/jandedobbeleer/oh-my-posh/src/properties"
+	"github.com/jandedobbeleer/oh-my-posh/src/runtime"
+)
+
+type NixShellType string
+
+const (
+	PURE    NixShellType = "pure"
+	IMPURE  NixShellType = "impure"
+	UNKNOWN NixShellType = "unknown"
+	NONE    NixShellType = "none"
+)
+
+type NixShell struct {
+	props properties.Properties
+	env   runtime.Environment
+
+	ShellType string
+}
+
+func (n *NixShell) Template() string {
+	return "via {{ .ShellType }}-shell"
+}
+
+func (n *NixShell) DetectShellType() NixShellType {
+	switch n.env.Getenv("IN_NIX_SHELL") {
+	case "pure":
+		return PURE
+	case "impure":
+		return IMPURE
+	default:
+		if n.InNewNixShell() {
+			return UNKNOWN
+		}
+		return NONE
+	}
+}
+
+// Hack to detect if we're in a `nix shell` (in contrast to a `nix-shell`).
+func (n *NixShell) InNewNixShell() bool {
```

Comment:
I don't think this offer any guarantees, it assumes the availability of one `PATH` entry implies we're in a nix shell whereas you could still have, and be running, `nix-shell` if that one's also on your `PATH`. Also, why is this relevant? If the shell type is there (`IN_NIX_SHELL `), do you need to do this validation?

Reply:
@JanDeDobbeleer so there are two types of shells by design, `nix shell` and `nix-shell`, some discussion can be found [here](https://discourse.nixos.org/t/nix-shell-nix-shell-and-nix-develop/25964).

When we use `nix shell` the `IN_NIX_SHELL` variable isn't set, see https://github.com/NixOS/nix/issues/6677 so we check the `PATH` for it to verify.

https://starship.rs/ also uses the same approach for detecting it.

Reply:
@Yash-Garg can you add that issue to the inline comments as well. I don't understand how validating a PATH entry even closely qualifies as implying being in a certain shell environment, but I guess that's just me.

Reply:
@JanDeDobbeleer sure, i've added the comments for it.